core/src/autogluon/core/searcher/local_random_searcher.py

```diff
@@ -31,12 +32,24 @@ def _get_params_space(self) -> dict:
                 param_space[key] = sk
         return param_space
 
+    def _get_num_configs(self) -> int:
+        num_unique = 1
+        for key, val in self.search_space.items():
+            if isinstance(val, Space):
+                if isinstance(val, DiscreteSpace):
+                    num_unique *= len(val)
+                else:
+                    num_unique = None
+                    break
+        return num_unique
```

Comment:
Currently we would simply set num_unique = None, which only impacts the log message the user receives about the search space being exhausted. So the impact is not very significant. We can revisit if we add nested spaces, likely it would just be a call to the nested space such as len(nested_space).